syntax = "proto3";

import "core.proto";

package catkit_core.service;

message GetServiceInfoRequest
{
}

message GetServiceInfoReply
{
    string service_name = 1;
    string service_type = 2;
    string config = 3;

    repeated string parameter_names = 4;
    repeated string command_names = 5;
    repeated string datastream_ids = 6;
}

message GetParameterRequest
{
    string parameter_name = 1;
}

message GetParameterReply
{
    Value parameter_value = 1;
}

message SetParameterRequest
{
    string parameter_name = 1;
    Value parameter_value = 2;
}

message SetParameterReply
{
    Value parameter_value = 1;
}

message ExecuteCommandRequest
{
    string command_name = 1;
    Dict arguments = 2;
}

message ExecuteCommandReply
{
    Value result = 1;
}
